### PR TITLE
add assertArrayContains and assertArrayNotContains

### PR DESCRIPTION
**Pull request by khru**
## 📚 Description

How to assert that we have something on an array

## 🔖 Changes

- added `assertArrayContains`
- added `assertArrayNotContains`

## ✅ To-do list

- [X] Make sure that all the pipeline passes
- [X] Make sure to update the `CHANGELOG.md` to reflect the new feature or fix

### CODE REVIEW

File: src/assert.sh [hunk 1]
```diff
@@ -135,3 +135,33 @@ function assertCommandNotFound() {
...
+  shift
+  local actual=("$@")
+  if ! [[ "${actual[*]}" == *"$expected"* ]]; then
+      ((_ASSERTIONS_FAILED++))
+      printFailedTest  "${label}" "${actual[*]}" "to contain" "${expected}"
+      return 1
+    fi
+
+    ((_ASSERTIONS_PASSED++))
+    return 0
```
Comment (Tito-Kati) [inline src/assert.sh:151]:
```suggestion
    ((_ASSERTIONS_FAILED++))
    printFailedTest  "${label}" "${actual[*]}" "to contain" "${expected}"
    return 1
  fi

  ((_ASSERTIONS_PASSED++))
  return 0
```

File: src/assert.sh [hunk 2]
```diff
@@ -135,3 +135,33 @@ function assertCommandNotFound() {
...
+  shift
+  local actual=("$@")
+  if [[ "${actual[*]}" == *"$expected"* ]]; then
+      ((_ASSERTIONS_FAILED++))
+      printFailedTest  "${label}" "${actual[*]}" "to not contain" "${expected}"
+      return 1
+    fi
+
+    ((_ASSERTIONS_PASSED++))
+    return 0
```
Comment (Tito-Kati) [inline src/assert.sh:166]:
```suggestion
    ((_ASSERTIONS_FAILED++))
    printFailedTest  "${label}" "${actual[*]}" "to not contain" "${expected}"
    return 1
  fi

  ((_ASSERTIONS_PASSED++))
  return 0
```

File: tests/unit/assert_test.sh [hunk 3]
```diff
@@ -155,4 +155,34 @@ function test_unsuccessful_assertCommandNotFound() {
...
+  assertSuccessfulCode "$(assertArrayContains "1234" "${distros[@]}")"
+}
+
+function test_unsuccessful_assertArrayContains() {
+  local distros=(Ubuntu 1234 Linux\ Mint)
+  assertEquals\
+      "$(printFailedTest "Unsuccessful assertArrayContains" "Ubuntu 1234 Linux Mint" "to contain" "a_non_existing_element")"\
+      "$(assertArrayContains "a_non_existing_element" "${distros[@]}")"
+
+  assertGeneralError "$(assertArrayContains "a_non_existing_element" "${distros[@]}")"
```
Comment (Tito-Kati) [inline tests/unit/assert_test.sh:171]:
```suggestion
  local distros=(Ubuntu 1234 Linux\ Mint)

  assertEquals\
    "$(printFailedTest "Unsuccessful assertArrayContains" "Ubuntu 1234 Linux Mint" "to contain" "a_non_existing_element")"\
    "$(assertArrayContains "a_non_existing_element" "${distros[@]}")"
  assertGeneralError "$(assertArrayContains "a_non_existing_element" "${distros[@]}")"
```

**Comment by khru** [comment]
don't know why the function fails on the static analysis tool step of the actions, that's why I disable it, if @Chemaclass or @Tito-Kati could help me I'd love that
```bash
# shellcheck disable=SC2317
function test_successful_assertArrayContains() {
  local distros=(Ubuntu 1234 Linux\ Mint)
  assertEquals "$SUCCESSFUL_EMPTY_MESSAGE" "$(assertArrayContains "1234" "${distros[@]}")"
  assertSuccessfulCode "$(assertArrayContains "1234" "${distros[@]}")"
}
```

**Review by Tito-Kati — CHANGES_REQUESTED** [review]
I see some issues with the code styles, but I think the functionality is great @khru . If you could accept the changes, I'd appreciate it.

When I get some time today, I'll take a look at the shellcheck issues you mentioned.

**Review by Tito-Kati — APPROVED** [review]

**Review by Chemaclass — APPROVED** [review]
Love it 💡

**Comment by khru** [comment]
@Tito-Kati I will merge and you can check the comment later

**Pull request merged**